File: src-tauri/src/model/knowledge_base.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct KnowledgeBase {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub embedding_model_id: Option<String>,
} 

File: src/api/knowledge_base.ts
import {APIAbc} from "./api.ts";

export type KnowledgeBase = {
    id: string,
    name: string,
    description: string,
    embedding_model_id: string,
    created_at: Date
}

export class KnowledgeBaseAPI extends APIAbc {
    protected tableName: string = 'knowledge_base';
}
